api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -173,10 +175,51 @@ private DbWorkspace lookupWorkspace(String workspaceNamespace) throws NotFoundEx
         workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
     workspaceService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
 
-    return ResponseEntity.ok(
-        leonardoMapper.toApiRuntime(
-            leonardoNotebooksClient.getRuntime(
-                workspaceNamespace, userProvider.get().getRuntimeName())));
+    try {
+      return ResponseEntity.ok(
+          leonardoMapper.toApiRuntime(
+              leonardoNotebooksClient.getRuntime(
+                  workspaceNamespace, userProvider.get().getRuntimeName())));
+    } catch (NotFoundException e) {
+      if (!workbenchConfigProvider.get().featureFlags.enableCustomRuntimes) {
+        throw e;
+      }
+
+      List<LeonardoListRuntimeResponse> runtimes =
+          leonardoNotebooksClient.listRuntimesByProject(workspaceNamespace);
+      if (runtimes.isEmpty()) {
+        throw e;
+      }
+
+      LeonardoListRuntimeResponse mostRecentRuntime =
+          runtimes.stream()
+              .sorted(
+                  (a, b) ->
+                      b.getAuditInfo()
+                          .getCreatedDate()
+                          .compareTo(a.getAuditInfo().getCreatedDate()))
+              .findFirst()
+              .get();
+
+      final String OVERRIDE_LABEL =
+          LeonardoMapper.RUNTIME_CONFIGURATION_TYPE_ENUM_TO_STORAGE_MAP.get(
+              RuntimeConfigurationType.USEROVERRIDE);
+      Map<String, String> runtimeLabels = (Map<String, String>) mostRecentRuntime.getLabels();
+      if (runtimeLabels != null
+          && OVERRIDE_LABEL.equals(runtimeLabels.get(LeonardoMapper.RUNTIME_LABEL_AOU_CONFIG))) {
+        Runtime runtime = leonardoMapper.toApiRuntime(mostRecentRuntime);
+        if (!runtime.getStatus().equals(RuntimeStatus.DELETED)) {
+          log.warning(
+              "Runtimes returned from ListRuntimes should be DELETED but found "
+                  + runtime.getStatus());
+        }
+
+        return ResponseEntity.ok(
+            leonardoMapper.toApiRuntime(mostRecentRuntime).status(RuntimeStatus.DELETED));
```

Comment:
I don't know if there's an elegant way of doing this, but if we encounter an old and invalid runtime, it would be nice to bail here and just return a 404. I'm not sure if we can expect that manifest as an NPE out of the mapper, or perhaps as a malformed runtime.